### PR TITLE
Fence usage of new Spicy API with version check.

### PR DESCRIPTION
**Pull request by rsmmr**

### CODE REVIEW

**Review by bbannier — CHANGES_REQUESTED** [review]

File: src/plugin/plugin.cc [hunk 1]
```diff
@@ -574,10 +574,12 @@ void plugin::Zeek_Spicy::Plugin::InitPostScript() {
 
     hilti::rt::configuration::set(hilti_config);
 
+#if SPICY_VERSION_NUMBER >= 10700
```
[the remark this answers is not in the document]
Comment (rsmmr) [reply src/plugin/plugin.cc:577]: I missed that, we can fence that too.

**Review by bbannier — DISMISSED** [review]
This builds fine now, but still fails to test successfully against the latest Spicy release 1.6.1,

```
[ #2] zeek.spicyz-paths ... failed
  % 'TEST_DIFF_CANONIFIER= btest-diff files' failed unexpectedly (exit code 1)
  % cat .diag
  == File ===============================
  cmake
  cmake/FindSpicy.cmake
  cmake/FindZeek.cmake
  cmake/ZeekSpicyAnalyzerSupport.cmake
  include
  include/zeek-spicy
  include/zeek-spicy/autogen
  include/zeek-spicy/autogen/config.h
  include/zeek-spicy/cookie.h
  include/zeek-spicy/debug.h
  include/zeek-spicy/driver.h
  include/zeek-spicy/file-analyzer.h
  include/zeek-spicy/packet-analyzer.h
  include/zeek-spicy/plugin
  include/zeek-spicy/plugin.h
  include/zeek-spicy/plugin/cookie.h
  include/zeek-spicy/plugin/debug.h
  include/zeek-spicy/plugin/file-analyzer.h
  include/zeek-spicy/plugin/packet-analyzer.h
  include/zeek-spicy/plugin/plugin.h
  include/zeek-spicy/plugin/protocol-analyzer.h
  include/zeek-spicy/plugin/runtime-support.h
  include/zeek-spicy/plugin/zeek-compat.h
  include/zeek-spicy/plugin/zeek-reporter.h
  include/zeek-spicy/protocol-analyzer.h
  include/zeek-spicy/runtime-support.h
  include/zeek-spicy/spicy-compat.h
  include/zeek-spicy/zeek-compat.h
  include/zeek-spicy/zeek-reporter.h
  spicy
  spicy/zeek.spicy
  spicy/zeek_file.spicy
  spicy/zeek_rt.hlt
  tests
  tests/Scripts
  tests/Scripts/canonify-zeek-log
  tests/Scripts/canonify-zeek-log-sorted
  tests/Scripts/diff-remove-abspath
  tests/Scripts/diff-remove-timestamps
  tests/Scripts/diff-sort
  tests/Scripts/spicy-version
  tests/Scripts/zeek-version
  tests/random.seed
  == Diff ===============================
  --- /tmp/test-diff.26248.files.baseline.tmp	2023-01-30 15:45:09
  +++ /tmp/test-diff.26248.files.tmp	2023-01-30 15:45:09
  @@ -6,7 +6,13 @@
   include/zeek-spicy
   include/zeek-spicy/autogen
   include/zeek-spicy/autogen/config.h
  +include/zeek-spicy/cookie.h
  +include/zeek-spicy/debug.h
  +include/zeek-spicy/driver.h
  +include/zeek-spicy/file-analyzer.h
  +include/zeek-spicy/packet-analyzer.h
   include/zeek-spicy/plugin
  +include/zeek-spicy/plugin.h
   include/zeek-spicy/plugin/cookie.h
   include/zeek-spicy/plugin/debug.h
   include/zeek-spicy/plugin/file-analyzer.h
  @@ -16,6 +22,11 @@
   include/zeek-spicy/plugin/runtime-support.h
   include/zeek-spicy/plugin/zeek-compat.h
   include/zeek-spicy/plugin/zeek-reporter.h
  +include/zeek-spicy/protocol-analyzer.h
  +include/zeek-spicy/runtime-support.h
  +include/zeek-spicy/spicy-compat.h
  +include/zeek-spicy/zeek-compat.h
  +include/zeek-spicy/zeek-reporter.h
   spicy
   spicy/zeek.spicy
   spicy/zeek_file.spicy
  =======================================

  % cat .stderr

[ #1] zeek.parse-error ... failed
  % 'btest-diff dpd.log' failed unexpectedly (exit code 1)
  % cat .diag
  == File ===============================
  == Diff ===============================
  --- /tmp/test-diff.26287.dpd.log.baseline.tmp	2023-01-30 15:45:10
  +++ /tmp/test-diff.26287.dpd.log.tmp	2023-01-30 15:45:10
  @@ -1,10 +0,0 @@
  -#separator \x09
  -#set_separator	,
  -#empty_field	(empty)
  -#unset_field	-
  -#path	dpd
  -#open XXXX-XX-XX-XX-XX-XX
  -#fields	ts	uid	id.orig_h	id.orig_p	id.resp_h	id.resp_p	proto	analyzer	failure_reason
  -#types	time	string	addr	port	addr	port	enum	string	string
  -XXXXXXXXXX.XXXXXX	CHhAvVGS1DHFjwGM9	192.150.186.169	49244	131.159.14.23	22	tcp	SPICY_SSH	failed to match regular expression (<...>/test.spicy:9:15) [SSH-2.0-OpenSSH_3.8.1p1\x0a]
  -#close XXXX-XX-XX-XX-XX-XX
  =======================================

  % cat .stderr
  [zeek] Registering TCP protocol analyzer spicy::SSH with Zeek
  [zeek]   Scheduling analyzer for port 22/tcp
  [zeek] Done with post-script initialization
  [zeek] confirming protocol 115/0
  1150485513.396389 analyzer error in /Users/bbannier/src/spicy-plugin/tests/.tmp/zeek.parse-error/test.spicy, line 9: failed to match regular expression
  [zeek] Shutting down Spicy runtime
  cat: dpd.log: No such file or directory

2 of 2 tests failed
```

**Comment by bbannier** [comment]
> ok, we could wrap the tests too, but not sure that's worth the trouble, unless I'm missing something?

If users install spicy-plugin with zkg it will run tests by default, so it is not a good look.

**Comment by rsmmr** [comment]
Alright, I disabed those two tests for < 1.7.

**Review by bbannier — APPROVED** [review]

**Pull request merged**